src/tools/pub/dependencies.md

```diff
@@ -235,32 +223,62 @@ Instead, the typical workflow is:
-constraints](/tools/pub/glossary#version-constraint) for your dependencies. You
-typically want to use the latest versions of the dependencies when you first
-create your app. Then you'll create and check in a
-[lockfile](/tools/pub/glossary#lockfile) that pins your dependencies to those specific
-versions. Specifying version constraints in your pubspec then is usually
-redundant (though you can do it if you want).
-
-For a [library package](/tools/pub/glossary#library-package) that you want users to
-reuse, though, it is important to specify version constraints. That lets people
+Specifying version constraints lets people
 using your package know which versions of its dependencies they can rely on to
 be compatible with your library. Your goal is to allow a range of versions as
 wide as possible to give your users flexibility. But it should be narrow enough
 to exclude versions that you know don't work or haven't been tested.
 
-The Dart community uses [semantic versioning][]<sup id="fnref:semver"><a
+The Dart community uses semantic versioning<sup id="fnref:semver"><a
 href="#fn:semver">1</a></sup>, which helps you know which versions should work.
 If you know that your package works fine with `1.2.3` of some dependency, then
-semantic versioning tells you that it should work (at least) up to `2.0.0`.
+semantic versioning tells you that it should work with any subsequent
+stable release before `2.0.0`.
+For details on pub's version system,
+see the [package versioning page](/tools/pub/versioning#semantic-versions).
+
+You can express version constraints using either
+_caret syntax_ (`^1.2.3`) or
+_traditional syntax_ (`'>=1.2.3 <2.0.0'`).
+
+### Caret syntax
+
+_Caret syntax_ is a compact way of expressing the most common
+sort of version constraint.
+`^version` means _the range of all versions guaranteed to be backwards
+compatible with the specified version_.
+
+For example, `^1.2.3` is equivalent to `'>=1.2.3 <2.0.0'`, and
+`^0.1.2` is equivalent to `'>=0.1.2 <0.2.0'`.
+The following is an example of caret syntax:
+
+{% prettify yaml tag=pre+code %}
+dependencies:
+  path: ^1.3.0
+  collection: ^1.1.0
+  string_scanner: ^0.1.2
+{% endprettify %}
+
+Because caret syntax was introduced in Dart 1.8.3,
+it requires an [SDK constraint][]
+(using [traditional syntax](#traditional-syntax))
+to ensure that older versions of pub don't try to process it.
+For example:
+
+{% prettify yaml tag=pre+code %}
+environment:
+  sdk: '>=1.8.3 <3.0.0'
+{% endprettify %}
 
-A version constraint is a series of:
+### Traditional syntax
+
+A version constraint that uses _traditional syntax_
+is a series of the following:
 
 `any`
 : The string `any` allows any version. This is equivalent to an empty
-  version constraint, but is more explicit. **While `any` is allowed,
-  we do not recommend it for performance reasons.**
+  version constraint, but is more explicit. **Although `any` is allowed,
+  we don't recommend it for performance reasons.**
```

Comment:
Unrelated, and let's not open the can of worms here. But I don't think "performances reasons" is the argument we should be using :)